src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_acl_permission.erl

```diff
@@ -69,13 +70,26 @@ validate_request('PUT', Req, #base_state{chef_db_context = DbContext,
     Body = wrq:req_body(Req),
     Ace = chef_json:decode_body(Body),
     Part = list_to_binary(wrq:path_info(acl_permission, Req)),
-    case chef_object_base:strictly_valid(acl_spec(Part), [Part], Ace) of
-        ok ->
-            oc_chef_wm_acl:validate_authz_id(Req, State,
-                                             AclState#acl_state{acl_data = Ace},
-                                             Type, OrgId, DbContext);
-        Other ->
-            throw(Other)
+    %% Make sure we have valid json before trying other checks
+    %% Throws if invalid json is found
+    check_json_validity(Part, Ace),
+    %% validate_authz_id will populate the ACL AuthzId, which is needed
+    %% for checking the ACL constraints; we need to run validate_authz_id
+    %% anyway, so go ahead and do so
+    {Req1, State1 = #base_state{resource_state = #acl_state{
+          authz_id = AuthzId}}} =
+                                  oc_chef_wm_acl:validate_authz_id(Req, State,
+                                                                   AclState#acl_state{acl_data = Ace},
+                                                                    Type, OrgId,  DbContext),
+
+    %% Check if we're violating any constraints around modifying ACLs
+    %% i.e. deleting default groups, etc.
+    case oc_chef_authz_acl_constraints:check_acl_constraints(AuthzId, Type, Part, Ace) of
```

Comment:
Good question. With great power comes great responsibility and all that jazz, but I'm dubious if we always need to give the superuser the power to route around everything in the system.

That said, we probably should be consistent. Do we allow the superuser to by pass other checks like this?